apollo-gradle-plugin/src/main/groovy/com/apollostack/android/gradle/ApolloIRgenTask.groovy

```diff
@@ -10,20 +10,22 @@ import org.gradle.api.tasks.util.PatternSet
 public class ApolloIRGenTask extends NodeTask {
   private static final String APOLLO_CODEGEN = "node_modules/apollo-codegen/lib/cli.js"
   private static final String DEFAULT_OUTPUT_DIR = "src/main/graphql"
-
+  protected static final String DEFAULT_SCHEMA_FILE_PATTERN = "**/schema.json"
   static final String NAME = "generate%sApolloIR"
-  List<ApolloExtension> config
+
   String variant
+  List<ApolloExtension> config
+  private List<String> possibleGraphQLPaths
 
   @OutputDirectory
-  /** Output directory for the generated IR, default to src/main/graphql **/
+  /** Output directory for the generated IR, defaults to src/main/graphql **/
   File outputDir
 
   @InputFiles
   Collection<File> getInputFiles() {
     Set<File> inputFiles = Sets.newHashSet()
     config.each { ext ->
-      ext.getFiles().asMap().entrySet().each { entry ->
+      ext.getFiles(ext.graphQLPath ?: "src/${ext.getSourceSet()}/graphql").asMap().entrySet().each { entry ->
```

Comment:
If a user specified a `graphQLPath` for a sourceSet, then this would take precedence over `src/sourceSetName/graphql`.